File: practice_projects/project euler/evenfib.js
function evenfib(first_value, second_value, lim) {
    let fib = [first_value, second_value]
    let result = 0
    let limit = lim
    // console.log(fib[fib.length - 2])

    while (true) {
        let last = fib[fib.length - 1]
        let snd_last = fib[fib.length - 2]
        let sum = last + snd_last

        fib.push(sum)

        if (sum >= limit) {
            break
        }
    }

    for (let i = 0; i < fib.length - 1; i++) {
        if (fib[i] % 2 == 0) {
            result += fib[i]
        }
    }

    console.log(result)
}

evenfib(1, 2, 4000000)

File: practice_projects/project euler/evenfib.py
def evenfib(first_value,second_value,lim):
    fib = [first_value,second_value]
    
    result = 0
    limit = lim
    
    while True:
        fib.append(fib[-1]+fib[-2])
        
        if fib[-1] >= limit:
            break
        
    for i in range(0,len(fib)-1):
        if fib[i]%2==0:
            result += fib[i]
            
    print(result)
    
evenfib(1,2,4000000)